src/ui/flutter_app/lib/services/storage/adapters/algorithm_adapter.dart

```diff
@@ -0,0 +1,46 @@
+/*
+  This file is part of Sanmill.
+  Copyright (C) 2019-2021 The Sanmill developers (see AUTHORS file)
+
+  Sanmill is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Sanmill is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import 'package:hive_flutter/adapters.dart';
+import 'package:json_annotation/json_annotation.dart' show JsonSerializable;
+import 'package:sanmill/models/preferences.dart';
+
+/// Algorithm Adapter
+///
+/// This adapter provides helper functions to be used with [JsonSerializable]
+/// and is a general [TypeAdapter] to be used with Hive [Box]ses
+class AlgorithmAdapter extends TypeAdapter<Algorithms> {
+  @override
+  final typeId = 8;
```

Comment:
Why typeId = 8? Magic number?

Reply:
hive type id's still aren't magic numbers. It's the same as `@HiveField(2)`. 
FYI this is an outdated commit. I've removed it since then and the class is only responsible for the json functions